### ISSUE TITLE
Why are there no events for `Seconded` and `Voted` in pallet democracy

### ISSUE THREAD

**Issue by JoshOrndorff** [paritytech/substrate#10283]
There are many events in the pallet as I would expect. However there are no events for the most common user actions, seconding a proposal, and casting a vote.

This makes it hard for off-chain tools such as UIs work with the chain. For example Subscan works around this by scanning for `vote` and `second` extrinsics performing these two actions.

However, it is not a valid assumption that all votes or seconds will be directly caused by an extrinsic. For example, imagine a runtime with another pallet that participates in democracy by calling its function. For example, Something similar to a treasury, but it votes with its tokes according to some rules.

Very concretely, in Moonbeam, we expose democracy to our EVM users through a pre-compile. Metamask users can use these precompiles, and so can real smart contracts, like a DAO for example. So when a vote or second is made through the precompile, tools like subscan are not able to detect the vote, and display incorrect results.

Proposed Solution: Add events, `Seconded` and `Voted` to pallet democracy.

**Comment by JoshOrndorff** [comment]
The Moonbeam team is willing to make a PR for this if the solution is acceptable.

**Comment by bkchr** [comment]
CC @thiolliere @shawntabrizi

**Comment by kianenigma** [comment]
> However, it is not a valid assumption that all votes or seconds will be directly caused by an extrinsic.

Sounds reasonable to me.

**Comment by JoshOrndorff** [comment]
@kianenigma My wording was not very precise; let me try again. Every vote or second will be initially caused by _some_ extrinsic. But they will not necessarily be caused by calls directly to pallet-democracy's own extrinsics. For example, in Moonbeam, we have an evm precompile that allows EVM accounts to vote by calling into the EVM. This is useful for users who are using metamask which only knows how to sign ethereum-style transactions. It is also useful for in-evm DAOs to participate in democracy. If you want to inspect further, look into https://github.com/PureStake/moonbeam/tree/master/precompiles/pallet-democracy

Concretely, what we are experiencing is that many metamask users are voting on our referenda through the EVM, and subscan cannot count those votes because subscan is scraping the chain for pallet_democracy.vote extrinsics.

If we add events for voting, subscan can use the events instead of scraping extrinsics, and our calls through the evm will also cause the events to be emitted.

**Comment by thiolliere** [comment]
adding a seconded and voted event is good to me, I think we should accept such a PR
